operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentConfig.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.operator.controllers;
+
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.ExecActionBuilder;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import org.keycloak.common.util.CollectionUtil;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatusBuilder;
+import org.keycloak.operator.crds.v2alpha1.deployment.ValueOrSecret;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Configuration for the KeycloakDeployment
+ */
+public class KeycloakDeploymentConfig {
```

Comment:
I would maybe make this normal "stateful" class and get rid of the direct dependency on `KeycloakDeployment` as I believe it doesn't belong here. Instead, I would suggest to create fields (and constructor) for Keycloak CR, StatefulSet and K8s Client.

This would allow us to test it better by unit tests (please add some for the Enabled/Disabled Features – just testing the CR values are really mapped to correct env vars) and will get rid of the ugly parameters in each method for passing `KeycloakDeployment`.

WDYT?

Reply:
Yep, it'll probably be better as it's associated with the KeycloakDeployment.